test/e2e/network_segmentation.go

```diff
@@ -875,12 +875,13 @@ var _ = Describe("Network Segmentation", func() {
 					cidr:     correctCIDRFamily(userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
 					role:     "primary",
 				}),
-				ginkgo.Entry("with primary layer2 UDN", networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer2",
-					cidr:     correctCIDRFamily(userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				}),
+				// TODO: this test is broken, see https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5309
```

Comment:
NIT: To clarify, I think the test with the changes in your PR is fine, it's just that there is a bug in l2 multicast :)